src/OrbitBase/include/OrbitBase/StopToken.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_STOP_TOKEN_H_
+#define ORBIT_BASE_STOP_TOKEN_H_
+
+#include <memory>
+
+#include "OrbitBase/SharedState.h"
+
+namespace orbit_base {
+
+// Forward declaration for befriending
+class StopSource;
```

Comment:
You can combine that with the friend declaration below and just write `friend class StopSource`.

Reply:
done